packages/core/src/plugins/rsdoctor.ts

```diff
@@ -43,25 +43,13 @@ export const pluginRsdoctor = (): RsbuildPlugin => ({
         return;
       }
 
-      let isAutoRegister = false;
-
-      for (const config of bundlerConfigs) {
-        const registered = config.plugins?.some(
-          (plugin) => plugin?.constructor?.name === pluginName,
-        );
-
-        if (registered) {
-          return;
-        }
-
-        config.plugins ||= [];
-        config.plugins.push(new module[pluginName]());
-        isAutoRegister = true;
+      if (chain.plugins.has(CHAIN_ID.PLUGIN.RSDOCTOR)) {
+        return;
       }
 
-      if (isAutoRegister) {
-        logger.info(`${color.bold(color.yellow(packageName))} enabled.`);
-      }
+      chain.plugin(CHAIN_ID.PLUGIN.RSDOCTOR).use(module[pluginName]);
```

Comment:
This may be a breaking change for appending the Rsdoctor plugin like this:

```js
mport { RsdoctorRspackPlugin } from '@rsdoctor/rspack-plugin';

export default {
  tools: {
    rspack(config, { appendPlugins }) {
      if (process.env.RSDOCTOR === 'true') {
        appendPlugins(
          new RsdoctorRspackPlugin({
            // plugin options
          }),
        );
      }
    },
  },
};
```

Reply:
This is an unexpected breaking change, we should allow users to define Rsdoctor plugin via `tools.rspack`, that's why Rsbuild uses the `onBeforeCreateCompiler` hook to check Rspack config and enable the Rsdoctor plugin.

Reply:
That's reasonable. I would find another way to modify the options of Rsdoctor. Thanks!